packages/central-server/src/apiV2/authenticate/authenticate.js

```diff
@@ -108,23 +107,41 @@ const checkApiClientAuthentication = async req => {
  * and if valid, returns a new JWT token that can be used for accessing the API
  * Override grants to do recursive authentication, for example when creating a new user.
  */
+
 export async function authenticate(req, res) {
   await req.assertPermissions(allowNoPermissions);
+  const knexInstance = req.database.connection;
 
-  const { refreshToken, user, accessPolicy } = await checkUserAuthentication(req);
-  const { user: apiClientUser } = await checkApiClientAuthentication(req);
+  const rateLimiter = new TupaiaRateLimiter(knexInstance);
 
-  const permissionGroupsByCountryId = await extractPermissionGroupsIfLegacy(
-    req.models,
-    accessPolicy,
-  );
-  const authorizationObject = await getAuthorizationObject({
-    refreshToken,
-    user,
-    accessPolicy,
-    apiClientUser,
-    permissionGroups: permissionGroupsByCountryId,
-  });
-
-  respond(res, authorizationObject);
+  if (await rateLimiter.checkIsRateLimited(req)) {
```

Comment:
I just saw this as I was a bit bogged down with build errors. I did think about this and assumed that we would want to rate limit those onetimelogin and refreshToken in the same way but it would be easy enough to configure them to have different logic if we want. Let's discuss later in the week.